R/pkg/R/serialize.R

```diff
@@ -37,24 +37,38 @@ writeObject <- function(con, object, writeType = TRUE) {
   # passing in vectors as arrays and instead require arrays to be passed
   # as lists.
```

Comment:
Thanks, @davies. 
To avoid the Warning, I have to add "type %in% check".
As you suggest, it still throws like "otherwise it will throw something like "During startup - Warning message:
In is.na(object) :
is.na() applied to non-(list or vector) of type 'environment'".